### ISSUE TITLE
Streaming / HLS / Zap.Stream support

### ISSUE THREAD

**Issue by alltheseas** [nostrability/nostrability#83]
**app**|**HLS support**|**Zap.stream support**|**B**|**C**|**D:**|**E:**|**comment**
-----|-----|-----|-----|-----|-----|-----|-----
damus|✅|❌|.| | | | 
primal|❌|?|.| | | | 
amethyst|?|?|?| | | | 
coracle|?|?|?| | | | 
nostrudel|?|?|.| | | | 
nostur|?|?|?| | | | 
voyage|?|?|?| | | | 
gossip|?|?|?| | | | 
openvibe|?|?|?| | | | 
lume|?|?|?| | | | 
ditto|?|?|?| | | | 
nostter|?|?|?| | | | 
nosotros|?|?|?| | | | 
highlighter|?|?|?| | | | 
yana|?|?|?| | | | 
nostrmo|?|?|?| | | | 
comingle|?|?|@tyiu| | | | 
nostrSDK|?|?|@tyiu| | | | 
Iris|?|?|@mmalmi| | | | 
wikifreedia|?|?|?| | | | 
wikistr|?|?|?| | | | 
yakihonne|?|?|?| | | | 
snort|?|?|?| | | | 
zap.stream|?|?|?| | | | 
nostr,nests|?|?|?| | | | 
dtan.xyz|?|?|?| | | | 
fountain|?|?|?| | | | 
wavlake|?|?|?| | | | 
NDK|?|?|?| | | | 
shopstr|?|?|?| | | |

https://njump.me/nevent1qqsp5rfq2mdfq7dpa5zu3fn9znqmgzqzfq46nr9snqwehgra4y9arngpzpmhxue69uhkummnw3ezuamfdejszrthwden5te0dehhxtnvdakqzxrhwvaz7te3xqczude39ceryd3wxycnqw358q6rsqg5waehxw309aex2mrp0yhxgctdw4eju6t02pzc5a

**Comment by alltheseas** [comment]
@v0l what standard does zapstream use for streaming?

**Comment by v0l** [comment]
[NIP-53](https://github.com/nostr-protocol/nips/blob/master/53.md)

**Comment by alltheseas** [comment]
@v0l do the apps you listed support viewing zapstream/nip-53 feeds only, or streaming + viewing?

**Comment by v0l** [comment]
Viewing / Zapping / Commenting only afaik

**Comment by alltheseas** [comment]
Are there any other apps beyond zapstream that can stream NIP-53?

**Comment by fiatjaf** [comment]
I don't know if it makes sense to have this giant table of unrelated clients.

**Comment by alltheseas** [comment]
> I don't know if it makes sense to have this giant table of unrelated clients.


what do you suggest?

**Comment by fiatjaf** [comment]
> what do you suggest?

having only clients that matter.

**Comment by alltheseas** [comment]
updated to a table with respondees only. is this good for now?

**Comment by YawningGoat** [comment]
**Zap.Stream**
- HLS - No (https://zap.stream/nevent1qqsrwv3a0ftfsy55jqaec8wzw27j8dn4k6r46834d65ftjcxgfp6c9cg8sn26)
- .webm - No (https://zap.stream/nevent1qqsfmxjnuh9awg9kjw7v6g64v3lpk6aaesfsg4ywery3auu8egyvghc8wmswu)

**Primal**
- View NIP-53 stream: No (on iOS and web)
- .webm Support: Yes (https://primal.net/e/note1nkd98ewt6ustdyaue5342er7rd4mmnqnq32gajxfrmec0jsgc30s95yuvh)
- Create NIP-53 stream: No (on iOS and web)

**Nostrudel**
- .webm - Yes
- Create NIP-53 stream: No 

**Nostur**
- .webm - No
- Create NIP-53 stream: No

**Coracle**
- HLS - No (streams audio only - https://coracle.social/notes/nevent1qyd8wumn8ghj7un9d3shjtn8v46xzmrz0yhxxmmd9amrzqgwwaehxw309ahx7uewd3hkctcpz9mhxue69uhkummnw3ezuamfdejj7qgkwaehxw309aex2mrp0yhx6mmnw3ezuur4vghsz9nhwden5te0v4jx2m3wdehhxarj9ekxzmny9uq32amnwvaz7tmjv4kxz7fwv3sk6atn9e5k7tcprdmhxue69uhhg6r9vehhyetnwshxummnw3erztnrdakj7qgkwaehxw309a482mncd9hxwampdenjummjvuhsqgphxg7h545cz22fqwuurhp890frke6mdp6arc6ka2y4evryysavzum5c4ax)
- View NIP-53 stream - No
- .webm - Yes (https://coracle.social/notes/nevent1qqsfmxjnuh9awg9kjw7v6g64v3lpk6aaesfsg4ywery3auu8egyvghc8wmswu)
- Create NIP-53 stream - No

**Snort:**
- HLS - No
- View NIP-53 steam - No
- .webm - No
- Create NIP-53 stream - No

Maybe I've misunderstood the . symbols you've used above (might be browser/system dependant, but I filled in what worked for me. :)

**Comment by alltheseas** [comment]
thank you @YawningGoat :pray: updated accordingly

@v0l wen streaming support in snort?? :smiley:

**Comment by fiatjaf** [comment]
This issue is tracking two completely independent things. NIP-53 livestream support is very different than just displaying HLS from a URL in a note. webm is yet another entirely different thing.

**Comment by alltheseas** [comment]
What do you think is a better approach?

**Comment by fiatjaf** [comment]
One issue for NIP-53 live video events, another for displaying webm, another for displaying HLS in notes.

But I would not try to track webm and HLS support in notes, these are not compatibility-critical issues, they are more like nice-to-have features of clients.

**Comment by alltheseas** [comment]
Specific to NIP-53, @tyiu is considering adding support to display nip-53 scheduled events on comingle, and potentially add a virtual event filter/toggle/ui element.

**Comment by alltheseas** [comment]
> In fact the fact that Damus is uploading video as HLS is kinda breaking everything

@fiatjaf upload damus default is .mp4. Damus renders HLS posted by any nostr app. For reference, here is the add support for streaming HLS issue from last year: https://github.com/damus-io/damus/issues/798

**Comment by alltheseas** [comment]
> One issue for NIP-53 live video events, another for displaying webm, another for displaying HLS in notes.

@fiatjaf what do you think about calling a new category of non-NIP feature tracker where HLS, webm could fit in: "not mandatory", "fyi", or "bonus points"

**Comment by alltheseas** [comment]
From the end user perspective the concept of "streaming" is involves more than NIP-53. 

See https://njump.me/note1vkulcuqym2whk2jkltcumevkw9eypmv459vkpyvq9ctzw97unx6sgrgzcu

I think its useful context to have for the streamer persona. Maybe nostrability repo is not the best place for it.

Do yall know of a nostr for streamers resource?

cc @fiatjaf

**Comment by jb55** [comment]
> In fact the fact that Damus is uploading video as HLS is kinda breaking everything, I think it's better if a simpler video URL that everybody understands as the main URL in the note content, then use that `imeta` tag to have an optional HLS link -- so yes, I think we need a different issue to address that.

this is not true, damus does no such thing.

**Comment by alltheseas** [comment]
Primal and iOS Damus are investigating adding NIP-53 streaming

**Comment by alltheseas** [comment]
@hzrd149 does Applesauce support NIP-53 streaming?

**Comment by sondreb** [comment]
> nostria
> 
> https://nostr.eu/nevent1qqsqqqq2sarpqsumdfc4y22hfakfqw9hkehamgnhlwsr9lqztd9z09g4z6t7q
> 
> <img alt="Image" width="960" height="333" src="https://private-user-images.githubusercontent.com/64376233/524924572-a1ba415d-6172-4c14-baa1-5e9db325ccaa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQyMTk0NzksIm5iZiI6MTc3NDIxOTE3OSwicGF0aCI6Ii82NDM3NjIzMy81MjQ5MjQ1NzItYTFiYTQxNWQtNjE3Mi00YzE0LWJhYTEtNWU5ZGIzMjVjY2FhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzIyVDIyMzkzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM3NjliMDQ1ZjJiZGMwYmViZTEzMGUyNDE4ZmI3M2YxZWZkYzE3NTJhYjZkZWRmZWIyODkwYzZkNGVmOGZhZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._bcN1bTjh6q4vYRMY9pvwIGKOv8P0jxtkBV5l47hdXo">

Yes, it supports live events and meetings, but it's not listed as supported here.

Not "create nip-53 livestream", but does view HLS streams. Pretty sure it supports .webm too, have not verified.
